dali/pipeline/pipeline_test.cc

```diff
@@ -533,51 +524,30 @@ TYPED_TEST(PipelineTest, TestSeedSet) {
 }
 
 
-class PrefetchedPipelineTest : public GenericDecoderTest<RGB> {
- protected:
-  uint32_t GetImageLoadingFlags() const override {
-    return t_loadJPEGs + t_decodeJPEGs;
-  }
-
-  void SetUp() override {
-    DALISingleOpTest::SetUp();
-    batch_size_ = 5;
-    // set_batch_size(jpegs_.nImages());
-  }
+class PrefetchedPipelineTest : public DALITest {
+ public:
 
-  void CheckResults(Pipeline &pipe, int batch_size, int Iter) {
+  void CheckResults(Pipeline &pipe, int batch_size, int Iter, TensorList<CPUBackend> &data) {
     DeviceWorkspace ws;
     pipe.Outputs(&ws);
     ASSERT_EQ(ws.NumOutput(), 1);
     ASSERT_EQ(ws.NumInput(), 0);
     ASSERT_TRUE(ws.OutputIsType<GPUBackend>(0));
     TensorList<GPUBackend> &res1 = ws.Output<GPUBackend>(0);
+
+    TensorList<CPUBackend> res_cpu;
+    res_cpu.Copy(res1, cudaStream_t(0));
+    CUDA_CALL(cudaDeviceSynchronize());
+
+    auto res_cpu_view = view<uint8_t>(res_cpu);
+    auto data_view = view<uint8_t>(data);
     for (int j = 0; j < batch_size; ++j) {
-      this->VerifyDecode(
-          res1.template tensor<uint8>(j),
-          res1.tensor_shape(j)[0],
-          res1.tensor_shape(j)[1], (Iter * batch_size + j));
+      int data_idx = (Iter * batch_size + j) % data_view.shape.num_samples();
+      Check(res_cpu_view[j], data_view[data_idx]);
     }
   }
```

Comment:
Maybe this can be extracted to a utility function (copy to the cpu and compare with the reference)?

Reply:
Ok, it is already is. Can you use it inside `executor_test.cc`?